src/nimblepkg/jsonhelpers.nim

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) Dominik Picheta. All rights reserved.
+# BSD License. Look at license.txt for more info.
+
+import json
+
+proc newJObjectIfKeyNotExists(obj: JsonNode, key: string): JsonNode =
+  assert obj.kind == JObject
+  if not obj.hasKey(key):
+    let newObj = newJObject()
+    obj.add(key, newObj)
+    return newObj
+  else:
+    return obj[key]
+
+proc addIfNotExist*(obj: JsonNode, keys: varargs[string],
+                    val: JsonNode): JsonNode =
+  # If the path in the `obj` json tree described by `keys` does not exist create
+  # it, add the node `val` to it and return the added node, otherwise return the
+  # value of the existing object at the end of the path.
+
+  assert obj.kind == JObject
+  var obj = obj
+  for i in 0 ..< keys.len() - 1:
+    obj = obj.newJObjectIfKeyNotExists(keys[i])
+  if not obj.hasKey(keys[^1]):
+    obj.add(keys[^1], val)
+    return val
+  else:
+    return obj[keys[^1]]
+
+proc cleanUpEmptyObjects*(obj: JsonNode): JsonNode =
+  if obj.kind == JObject:
+    result = newJObject()
+    for key, value in obj:
+      var newValue = cleanUpEmptyObjects(value)
+      if newValue.kind notin {JObject, JArray} or newValue.len != 0:
+        result.add(key, newValue)
+  elif obj.kind == JArray:
+    result = newJArray()
+    for value in obj:
+      var newValue = cleanUpEmptyObjects(value)
+      if newValue.kind notin {JObject, JArray} or newValue.len != 0:
+        result.add(newValue)
+  else:
+    result = obj
+
+when isMainModule:
+  import unittest
+
+  test "bewJObjectIfKeyNotExists":
+    proc testProc(testedJson, key, expectedResult: string) =
+      let testedJson = parseJson(testedJson)
+      let expectedResult = parseJson(expectedResult)
+      let actualResult = newJObjectIfKeyNotExists(testedJson, key)
+      check actualResult == expectedResult
+
+    testProc("{}", "key", "{}")
+    testProc("{ \"key1\": \"value1\", \"key2\": {} }", "key3", "{}")
+    testProc("{ \"key1\": \"value1\", \"key2\": {} }", "key1", "\"value1\"")
+    testProc("{ \"key1\": \"value1\", \"key2\": { \"key3\": [ 2, 3, 5] } }",
```

Comment:
I don't understand what type-safe JSON means. I think that keys have to be removed for the empty JSON objects from `nimbledata.json`, because when a library is removed its entry should be removed.